src/azure-cli/azure/cli/command_modules/cdn/custom.py

```diff
@@ -71,6 +91,315 @@ def update_endpoint(instance,
+                transforms=transforms
+            ))
+    if match_variable == 'RequestUri':
+        return DeliveryRuleRequestUriCondition(
+            parameters=RequestUriMatchConditionParameters(
+                operator=operator,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
+    if match_variable == 'RequestBody':
+        return DeliveryRuleRequestBodyCondition(
+            parameters=RequestBodyMatchConditionParameters(
+                operator=operator,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
+    if match_variable == 'RequestScheme':
+        return DeliveryRuleRequestSchemeCondition(
+            parameters=RequestSchemeMatchConditionParameters(
+                match_values=match_values.split(","),
+                negate_condition=negate_condition
+            ))
+    if match_variable == 'UrlPath':
+        return DeliveryRuleUrlPathCondition(
+            parameters=UrlPathMatchConditionParameters(
+                operator=operator,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
+    if match_variable == 'UrlFileExtension':
+        return DeliveryRuleUrlFileExtensionCondition(
+            parameters=UrlFileExtensionMatchConditionParameters(
+                operator=operator,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
+    if match_variable == 'UrlFileName':
+        return DeliveryRuleUrlFileNameCondition(
+            parameters=UrlFileNameMatchConditionParameters(
+                operator=operator,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
+    if match_variable == 'HttpVersion':
+        return DeliveryRuleHttpVersionCondition(
+            parameters=HttpVersionMatchConditionParameters(
+                match_values=match_values.split(","),
+                negate_condition=negate_condition
+            ))
+    if match_variable == 'IsDevice':
+        return DeliveryRuleIsDeviceCondition(
+            parameters=IsDeviceMatchConditionParameters(
+                match_values=match_values.split(","),
+                negate_condition=negate_condition
+            ))
+    if match_variable == 'Cookies':
+        return DeliveryRuleCookiesCondition(
+            parameters=CookiesMatchConditionParameters(
+                operator=operator,
+                selector=selector,
+                match_values=match_values.split(","),
+                negate_condition=negate_condition,
+                transforms=transforms
+            ))
```

Comment:
why not use if .. elif ... else?

Reply:
Every if end with a return statement so elif is unnecessary. You will get a style error if you replace  if with elif.